src/renderer/api-decorator.js

```diff
@@ -560,52 +560,93 @@ limitations under the License.
      * Preload script eval
      */
     ipc.once(`post-api-injection-${renderFrameId}`, () => {
-        const identity = {
-            uuid: initialOptions.uuid,
-            name: initialOptions.name
-        };
-        let { preload: preloadOption, plugin: plugin } = convertOptionsToElectronSync(getWindowOptionsSync());
-        if (Array.isArray(plugin) && plugin[0] !== undefined) { preloadOption = preloadOption.concat(plugin); }
-        const action = 'set-window-preload-state';
+        const { uuid, name } = initialOptions;
+        const identity = { uuid, name };
+        const windowOptions = getWindowOptionsSync();
 
-        if (preloadOption.length) { // short-circuit
-            let response;
-            try {
-                response = syncApiCall('get-selected-preload-scripts', preloadOption);
-            } catch (error) {
-                logPreload('error', identity, 'error', '', error);
-            }
+        let { plugin, preload } = convertOptionsToElectronSync(windowOptions);
```

Comment:
is `convertOptionsToElectronSync` safe to call multiple times? Is it called elsewhere?

Reply:
it was there before the change... might change later on, not in this PR (phase 1)